package/MDAnalysis/core/universe.py

```diff
@@ -992,6 +993,194 @@ def add_Segment(self, **attrs):
         self.segments = SegmentGroup(np.arange(self._topology.n_segments), self)
         # return the new segment
         return self.segments[segidx]
+    
+    def _add_TopologyObjects(self, object_type, values, types=None, guessed=False,
+                           order=None):
+        """Add new TopologyObjects to this Universe
+
+        Parameters
+        ----------
+        object_type : {'bonds', 'angles', 'dihedrals', 'impropers'}
+            The type of TopologyObject to add.
+        values : iterable of tuples, AtomGroups, or TopologyObjects
+            An iterable of: tuples of atom indices, or AtomGroups, 
+            or TopologyObjects. If every value is a TopologyObject, all 
+            keywords are ignored.
+        types : types (optional)
```

Comment:
Is `types` a scalar, array-like, ... ? Not clear. (Also applies to docs below.)

It's especially confusing because `type` is also a Python thing.

Reply:
`types` can be either `None` or array-like. I think the LAMMPS parser actually uses them and I put them in ITPParser. When `types=None`, `TopologyObject.type` just returns the atom types of the atoms involved. `TopologyGroup.types()` returns an ordered unique list of every TopologyObject.type.

I'm not sure whether MDAnalysis uses the `type` information anywhere. 

```python
In [1]: import MDAnalysis as mda

In [2]: from MDAnalysis.tests.datafiles import *

In [3]: lmp = mda.Universe(LAMMPSdata)

In [4]: lmp.bonds[0].type
Out[4]: '3'

In [5]: lmp.bonds.types()
Out[5]: ['3', '5', '7', '2', '1', '4', '6']

In [6]: len(lmp.bonds)
Out[6]: 18336
```